.github/workflows/newbuild-vcpkg.yml

```diff
@@ -0,0 +1,333 @@
+          - os: "windows-2022"
+            triplet: "x64-windows"
+            mono: ""
+            sudo: ""
+            cmake_config_options: '-G "Visual Studio 17 2022" -T host=x64 -A x64 -DUSE_OPENLDAP=OFF -DUSE_CASSANDRA=OFF -DUSE_AZURE=OFF -DWSSQL_SERVICE=OFF -DINCLUDE_PLUGINS=OFF'
+            cmake_build_options: "--config Release -j2 -- -m"
+            event_name: "pull_request, push, schedule"
+          - os: "macos-11"
+            triplet: "x64-osx"
+            mono: "mono"
+            sudo: "sudo"
+            cmake_config_options: "-DCMAKE_BUILD_TYPE=Release -DINCLUDE_PLUGINS=ON -DSUPPRESS_V8EMBED=ON -DSUPPRESS_REMBED=ON -DSUPPRESS_MEMCACHED=ON -DSUPPRESS_MONGODBEMBED=ON -DUSE_JAVA=OFF"
+            cmake_build_options: "-- -j$(nproc) -k"
+            event_name: "pull_request, push, schedule"
+          - os: "macos-12"
+            triplet: "x64-osx"
+            mono: "mono"
+            sudo: "sudo"
+            cmake_config_options: "-DCMAKE_BUILD_TYPE=Release -DINCLUDE_PLUGINS=ON -DSUPPRESS_V8EMBED=ON -DSUPPRESS_REMBED=ON -DSUPPRESS_MEMCACHED=ON -DSUPPRESS_MONGODBEMBED=ON -DUSE_JAVA=OFF"
+            cmake_build_options: "-- -j$(nproc) -k"
+            event_name: "pull_request, push, schedule"
+      fail-fast: false
+
+    name: "Build with vcpkg"
+    needs: check-skip
+    if: github.repository == 'hpcc-systems/HPCC-Platform'
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Display build summary
+        working-directory: .
+        run: |
+          echo ${{ matrix.os }} ${{ matrix.triplet }}
+          echo "Checkout to $Env:GITHUB_WORKSPACE"
+          echo "Platform Changed:  ${{ needs.check-skip.outputs.platform }}"
+
+      - name: Ubuntu Dependencies
+        if: ${{ contains(matrix.event_name, github.event_name) && needs.check-skip.outputs.platform && contains(matrix.os, 'ubuntu') }}
+        shell: "bash"
+        run: |
+          sudo apt-get update -y
+          sudo apt-get install -yq --no-install-recommends \
+          apt-transport-https \
+          autoconf \
+          autoconf-archive \
+          automake \
+          autotools-dev \
+          binutils-dev \
+          bison \
+          build-essential \
+          ca-certificates \
+          curl \
+          dirmngr \
+          flex \
+          git \
+          gnupg \
+          groff-base \
+          libtool \
+          pkg-config \
+          software-properties-common \
+          tar \
+          unzip \
+          uuid-dev \
+          zip
+
+      - name: Ubuntu Mono Dependencies
+        if: ${{ matrix.os == 'ubuntu-22.04' }}
+        shell: "bash"
+        run: |
+          sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 3FA7E0328081BFF6A14DA29AA6A19B38D3D831EF
+          echo "deb https://download.mono-project.com/repo/ubuntu stable-focal main" | sudo tee /etc/apt/sources.list.d/mono-official-stable.list
+          sudo apt-get update -y
+          sudo apt-get -yq install -f mono-complete
```

Comment:
question: why is mono required for ubuntu-22.04 ?

Reply:
mono is needed to run nuget, which is needed to fetch the prebuilt dependencies from the NuGet server